src/textual/command.py

```diff
@@ -215,6 +274,22 @@ async def search(self, query: str) -> Hits:
         """
         yield NotImplemented
 
+    async def discover(self) -> Hits:
+        """A default collection of hits for the provider.
+
+        Yields:
+            Instances of [`DiscoveryHit`][textual.command.DiscoveryHit].
+
+        Note:
+            This is different from
+            [`search`][textual.command.Provider.search] in that it should
+            yield [`DiscoveryHit`s][textual.command.DiscoveryHit] that
+            should be shown by default; before user input.
```

Comment:
Maybe I just need to reread what the semicolon is used for, but all of the suggestions below read better to me than the current docstring.

```suggestion
            should be shown by default (before user input).
```
```suggestion
            should be shown before user input.
```
```suggestion
            should be shown by default.
```
```suggestion
            should be shown when there is no user input.
```